File: Libraries/ImageWebResources/resources/csharp-app/ImageDownloader.cs
using System;

namespace Interpreter.Libraries.ImageWebResources
{
    internal static class ImageDownloader
    {
        public static bool BytesToImage(object byteArrayObj, object[] output)
        {
            byte[] bytes = (byte[])byteArrayObj;
            try
            {
                System.Drawing.Bitmap image;
                using (System.IO.MemoryStream stream = new System.IO.MemoryStream(bytes))
                {
                    image = (System.Drawing.Bitmap)System.Drawing.Bitmap.FromStream(stream);
                }
                output[0] = image;
                output[1] = image.Width;
                output[2] = image.Height;
            }
            catch (Exception)
            {
                return false;
            }

            return true;
        }
    }
}

File: Libraries/ImageWebResources/native/java-app/source/ImageDownloader.java
package org.crayonlang.libraries.imagewebresources;

import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import javax.imageio.ImageIO;
import org.crayonlang.interpreter.structs.*;

public class ImageDownloader {

	public static boolean bytesToImage(Object byteArrayObj, Object[] output) {
		byte[] bytes = (byte[]) byteArrayObj;
		ByteArrayInputStream inputStream = new ByteArrayInputStream(bytes);
		BufferedImage image = null;
		try {
			image = ImageIO.read(inputStream);
		} catch (java.io.IOException e) {
			return false;
		}
		
		output[0] = image;
		output[1] = image.getWidth();
		output[2] = image.getHeight();
		
		return true;
	}
}
